OpenBLAS build failing on mac

When trying to build OpenBLAS from scratch (on my Mojave Mac, using clang 10), rather than passing a pre-built version in a machine file, I get the following error:
```
Undefined symbols for architecture x86_64:
  "_LAPACKE_dgesvd", referenced from:
      ___ctest_kernel_regress_skx_avx_run in test_kernel_regress.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [openblas_utest] Error 1
make[3]: *** [tests] Error 2
```
@jeff-cohere has also seen another issue in this build relating to `ar` when using clang 12.

I'll note, also, that executing the build command that is used by cmake:
```
make -j NO_SHARED=1;PREFIX=/Users/mjschm/Sandia/collaboration/parallel_particles/cpp_particleCode/build;BUILD_DOUBLE=1
```
from the `ext/OpenBLAS` directory does successfully build OpenBLAS.

--- a/ext/CMakeLists.txt
+++ b/ext/CMakeLists.txt
@@ -271,7 +271,12 @@ if (HAERO_DEVICE STREQUAL "CPU" AND NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel")
   set_target_properties(openblas PROPERTIES IMPORTED_LOCATION ${OPENBLAS_LIBRARY_DIR}/libopenblas.a)
   if (NOT EXISTS ${OPENBLAS_LIBRARY})
     update_submodules()
-    set(OPENBLAS_OPTS NO_SHARED=1 PREFIX=${PROJECT_BINARY_DIR})
+    set(OPENBLAS_OPTS libs CC=${CMAKE_C_COMPILER}
+                      FC=${CMAKE_Fortran_COMPILER}
+                      NO_SHARED=1 PREFIX=${PROJECT_BINARY_DIR})
+    if (NOT APPLE)
+      set(OPENBLAS_OPTS ${OPENBLAS_OPTS} -j)
+    endif()
     if (HAERO_DOUBLE_PRECISION)
       set(OPENBLAS_OPTS ${OPENBLAS_OPTS} BUILD_DOUBLE=1)
     else()
@@ -283,9 +288,9 @@ if (HAERO_DEVICE STREQUAL "CPU" AND NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel")
                         BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/OpenBLAS
                         INSTALL_DIR ${PROJECT_BINARY_DIR}
                         CONFIGURE_COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/OpenBLAS ${CMAKE_CURRENT_BINARY_DIR}/OpenBLAS
-                        BUILD_COMMAND ${MAKE} -j ${OPENBLAS_OPTS}
+                        BUILD_COMMAND ${MAKE} ${OPENBLAS_OPTS}
                         LOG_BUILD TRUE
-                        INSTALL_COMMAND ${MAKE} -j ${OPENBLAS_OPTS} install
+                        INSTALL_COMMAND ${MAKE} ${OPENBLAS_OPTS} install
                         LOG_INSTALL TRUE)
 
     # Because the OpenBLAS people feel that life isn't complicated enough, they

